ssv/partial_sig_container.go

```diff
@@ -43,10 +29,44 @@ func (ps *PartialSigContainer) ReconstructSignature(root [32]byte, validatorPubK
 	return signature.Serialize(), nil
 }
 
-func (ps *PartialSigContainer) HasQuorum(root [32]byte) bool {
-	return uint64(len(ps.Signatures[rootHex(root)])) >= ps.Quorum
+// SignaturesForRoot returns a map of signer and signature for a specific root
+func (ps PartialSignatureContainer) SignaturesForRoot(root [32]byte) map[types.OperatorID][]byte {
+	sigs := make(map[types.OperatorID][]byte, 0)
+	for operatorID, sigMsg := range ps {
+		for _, msg := range sigMsg.Message.Messages {
+			if bytes.Equal(msg.SigningRoot[:], root[:]) {
+				sigs[operatorID] = msg.PartialSignature
+			}
+		}
+	}
+	return sigs
+}
+
+// Roots returns roots for the partial sigs
+func (ps PartialSignatureContainer) Roots() [][32]byte {
+	ret := make([][32]byte, 0)
+
+	if len(ps) <= 0 {
+		return ret
+	}
+
+	for _, sigMsg := range ps {
+		for _, msg := range sigMsg.Message.Messages {
+			ret = append(ret, msg.SigningRoot)
+		}
+		break // only need to iterate first msg
+	}
+	return ret
 }
 
-func rootHex(r [32]byte) string {
-	return hex.EncodeToString(r[:])
+// AllSorted returns ordered by signer array of signed messages
+func (ps PartialSignatureContainer) AllSorted() []*types.SignedPartialSignatureMessage {
+	ret := make([]*types.SignedPartialSignatureMessage, 0)
+	for _, m := range ps {
+		ret = append(ret, m)
+	}
+	sort.Slice(ret, func(i, j int) bool {
```

Comment:
for less memory allocations:
```suggestion
	ret := make([]*types.SignedPartialSignatureMessage, len(ps))
	for i, m := range ps {
		ret[i] = m
	}
```

Reply:
why delete sort?